src/components/progress-report/workflow/progress-report-workflow.service.ts

```diff
@@ -100,6 +126,84 @@ export class ProgressReportWorkflowService {
       this.repo.changeStatus(reportId, transition.to),
     ]);
 
-    // TODO(transition.notify);
+    const notificationInfo = await this.repo.getNotificationInfoByReportId(
+      reportId,
+    );
+    const projectId = notificationInfo[0]?.projectId ?? undefined;
+    const languageDisplayName = notificationInfo[0]?.languageDisplayName;
+    const reportQtrLabel = `Q${fiscalQuarter(
+      notificationInfo[0]?.start,
+    )} FY${fiscalYear(notificationInfo[0]?.start)}`;
+
+    let emailList: string[] = [];
+    notificationInfo.forEach((item) => {
+      item.role?.forEach((role) => {
+        if (transition.notify.TeamMembersWithRoles.includes(role)) {
+          if (item.email) emailList.push(item.email);
+        }
+      });
+    });
+
+    transition.notify.Emails.forEach((element: string) => {
+      if (element.includes('@')) {
+        emailList.push(element);
+      } else if (element === 'MONDAY_EMAIL_DISTRO') {
+        emailList.push(this.configService.mondayEmailDistro);
+      }
+    });
+
+    emailList = uniq(emailList);
+    const recipients = emailList.map((email) =>
+      this.prepareRecipientObject(email),
+    );
+
+    const changedBy = await this.userService.readOne(session.userId, session);
+    const project = await this.projectService.readOne(projectId, session);
```

Comment:
If the recipient is a specific user we should use a session corresponding to them instead of the user that is making the mutation.
That's what we do [here](https://github.com/SeedCompany/cord-api-v3/blob/cc00184941ece5480fc301c9bfbec308e27dac93/src/components/project/project.rules.ts#L1052-L1071).

Reply:
for my own learning, why is it necessary to get a session for each recipient?

Reply:
Basically these emails are UI for the user. All UI (or even data at a lower level) should be given under the lens of that user. So maybe a certain recipient can't see the project name, even though the user triggering the notification can. We wouldn't want to show the project name to that user just because another unrelated user is taking an action.
I know that's a bit contrived, but that's the idea. Also it's unlikely that this changes the actual data shown in the email for the users but the principle is important. And the more we do it, the more patterns will emerge to help form an easier way to do these kinds of things.